File: medium/1759.ts
/**
    # 1759. Count Number of Homogenous Substrings
 */
function countHomogenous(s: string): number {
    let ch: string = s[0];
    let len: number = 1;
    let res: number = 1;
    let i: number = 1;
    const MOD: number = 1e9 + 7;

    while (i < s.length) {
        if (s[i] == ch) {
            len = (len + 1) % MOD;
        }
        else {
            len = 1;
            ch = s[i];
        }

        res = (res + len) % MOD;
        ++i;
    }

    return res;
};

File: medium/1759.c
/**
    # 1759. Count Number of Homogenous Substrings
*/

#define MOD 1000000007

int countHomogenous(char * s){
    int len = strlen(s);
    char ch = *s;
    long sum = 1;
    long res = 1;

    for(int i = 1; i < len; ++i){
        if(ch == s[i]){
            sum = (sum + 1) % MOD;
        }
        else{
            ch = s[i];
            sum = 1;
        }
        res = (res + sum) % MOD;
    }

    return res;
}
